### ISSUE TITLE
[joss review] Algebra between operators defined on different grids

### ISSUE THREAD

**Issue by jakelangham** [jcorbino/mole#13]
While playing around with the MATLAB component of MOLE (as part of https://github.com/openjournals/joss-reviews/issues/6288), I noticed that it is possible to add or multiply operators that are constructed with respect to different grids. 

e.g. grad(2, 10, 1) + grad(2, 10, 2)

It also seems to be possible to add together operators on uniform and non-uniform grids. 

I'm not sure what this would mean exactly. Is ever desirable to be able to do this? If not then perhaps the code should throw an error in these cases.

**Comment by jcorbino** [comment]
You're correct @jakelangham . Non-unary operations involving operands constructed over different grids can be performed, and this is certainly not a justified scenario (not that I can think of). I don't think one can handle exceptions of that kind without defining the operators (currently simply sparse matrices) as objects of a class. This will be a fundamental change in the code, **which I'm not opposed to**, but it will certainly require a significant amount of time.

**Comment by jakelangham** [comment]
Yep, I suspected as much. Perhaps it might be beneficial to the library in the future, especially if there are other reasons why supplying additional class data with the operators turns out to be useful.

In the short term, I would recommend that the documentation is supplied with a warning about this. Of course it's not likely to cause users problems in simple scenarios, but in e.g. a more complex code testing various grid refinements, one could foresee it leading to bugs down the line.

**Comment by jcorbino** [comment]
@jakelangham - I added this to the README file:

**NOTE:** Performing non-unary operations involving operands constructed over different grids may lead to unexpected results. While MOLE currently allows such operations without throwing errors, users must exercise caution when manipulating operators across different grids.

And I also added that note to the manual.

I think the only pending issue now is this one: [https://github.com/jcorbino/mole/issues/12](https://github.com/jcorbino/mole/issues/12)
